### PR TITLE
fix(callouts): render nested callouts properly

### PR DESCRIPTION
**Pull request by Luke-zhang-04**
- fix(callouts): render nested callouts properly
  - previously, nested callouts were not rendered properly because regex could not detect the correct closing tag for a callout that had another callout nested inside of it.
- feat(callouts): make callouts look like Obsidian callouts
  - change callout look
  - make callouts collapsible (https://github.com/oleeskild/obsidian-digital-garden/issues/89)

Screen Shot 2023-01-11 at 17 51 01

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -65,13 +66,16 @@ module.exports = function(eleventyConfig) {
                     const code = token.content.trim();
                     if (code && code.toLowerCase().startsWith("title:")) {
                         const title = code.substring(6, code.indexOf("\n"));
-                        const titleDiv = title ? `<div class="admonition-title">${title}</div>` : '';
```
Comment (Luke-zhang-04) [inline .eleventy.js:68]: I haven't used the plugin or this style of admonition before so I don't know if I did this right

File: src/site/_includes/components/pageheader.njk [hunk 2]
```diff
@@ -1,5 +1,42 @@
 <meta name="viewport" content="width=device-width, initial-scale=1.0">
 <script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
+<script src="https://unpkg.com/lucide@0"></script>
+<script>
+    // Create callout icons
+    window.addEventListener("load", () => {
```
Comment (Luke-zhang-04) [inline src/site/_includes/components/pageheader.njk:6]: This is the questionable JS

**Comment by oyes77** [comment]
this pls, callouts now look really bad, or don't use the current theme styling

File: .eleventy.js (hunk 1 above)
Comment (oleeskild) [reply .eleventy.js:68]: This should be fine. Admonition-title was the internal class name used when callouts wasn't a thing and admonition was the way to go. I simply hadn't bothered to change it. "callout-title" is more correct as it is now.

File: .eleventy.js [hunk 3]
```diff
@@ -172,27 +176,46 @@ module.exports = function(eleventyConfig) {
 
 
     eleventyConfig.addTransform('callout-block', function(str) {
-        return str && str.replace(/<blockquote>((.|\n)*?)<\/blockquote>/g, function(match, content) {
-            let titleDiv = "";
-            let calloutType = "";
-            const calloutMeta = /\[!(\w*)\](\s?.*)/g;
-            if (!content.match(calloutMeta)) {
-                return match;
+        const parsed = parse(str);
```
Comment (oleeskild) [inline .eleventy.js:179]: Smart! Much cleaner and better than my attempt to regex HTML 😅

File: src/site/_includes/components/pageheader.njk (hunk 2 above)
Comment (oleeskild) [reply src/site/_includes/components/pageheader.njk:6]: I don't think this is questionable at all. It's clean, and gets the job done.

File: src/site/styles/style.scss [hunk 4]
```diff
@@ -92,3 +101,120 @@ p>code {
...
+.callout-title-inner {
+    font-weight: 600;
+}
+.callout-fold {
+    display: flex;
+    align-items: center;
+    padding-right: 8px;
+}
+.callout-fold .svg-icon {
+    transition: transform 100ms ease-in-out;
```
Comment (oleeskild) [inline src/site/styles/style.scss:216]: The transition is a nice touch 👌

**Review by oleeskild — COMMENTED** [review]
Looks very nice, thanks a lot for taking the time to create a PR for this. I've done some really minor refactors, and added the "svg-icon" class to the lucide svg elements, so that we are using the same styling as in Obsidian. Merging this and releasing right away.

**Comment by oleeskild** [comment]
The commits has been merged outside of GitHub

**Pull request closed without merging**